src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -602,6 +606,7 @@ public Task<Transaction> MergeTransactionSignaturesAsync(uint256 depositId, Tran
                                 {
                                     this.logger.LogInformation("Deposit: {0} collected enough signatures and is FullySigned", transfer.DepositTransactionId);
                                     transfer.SetStatus(CrossChainTransferStatus.FullySigned);
+                                    this.signals.Publish(new CrossChainTransferTransactionFullySigned(transfer));
```

Comment:
What's the advantage of using the event bus here vs just calling the method?

I think the event-driven approach is less useful because if there is a reorg we could end up broadcasting a tx that is no longer valid.

Reply:
mainly separation of concerns and the fact that now you have an event that you can subscribe to if you want to know everytime a transaction is fully signed.
Removing `async` from the subsriber handler would be literally the same of calling broadcast straight here, except that you don't have an explicit event to subscribe to if you need.

Btw even if you broadcast here directly you'll have the same problem of broadcasting a transaction in case of reorg, so I wouldn't be worried about it.

If you think this event will never been needed then we can remove it but I think it could be useful